include/boost/http/status_code-inl.hpp

```diff
@@ -121,6 +121,8 @@ String to_string(status_code sc)
         return "Not Extended";
     case status_code::network_authentication_required:
         return "Network Authentication Required";
+    default:
+        return "";
```

Comment:
The warning is:
```
    warning: control reaches end of non-void function [-Wreturn-type]
```
The proposed fix also disables the compiler warnings about unhandled enum values (`-Wswitch-enum`), which is something we do want.

I suggest that instead of returning an empty string from the default case, to return the empty string after the switch statement. That way we disable the return-type warning, but not the enum warning.